pkg/feature/raw_resources.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright (c) 2016-2017 Bitnami
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package feature
+
+import (
+	"context"
+	"fmt"
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"os"
+	"regexp"
+	"strings"
+)
+
+const (
+	YamlSeparator = "(?m)^---[ \t]*$"
+)
+
+type NameValue struct {
+	Name  string
+	Value string
+}
+
+func (f *Feature) createResourceFromFile(filename string, elems ...NameValue) error {
+	elemsMap := make(map[string]NameValue)
+	for _, nv := range elems {
+		elemsMap[nv.Name] = nv
+	}
+
+	data, err := os.ReadFile(filename)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	splitter := regexp.MustCompile(YamlSeparator)
+	objectStrings := splitter.Split(string(data), -1)
+	for _, str := range objectStrings {
+		if strings.TrimSpace(str) == "" {
+			continue
+		}
+		u := &unstructured.Unstructured{}
+		if err := yaml.Unmarshal([]byte(str), u); err != nil {
+			return errors.WithStack(err)
+		}
+
+		name := u.GetName()
+		namespace := u.GetNamespace()
+		if namespace == "" {
+			if val, exists := elemsMap["namespace"]; exists {
+				u.SetNamespace(val.Value)
+			} else {
+				u.SetNamespace("default")
```

Comment:
fall back to use "default" namespace?

Reply:
Yeah, this is actually how it was done in v1 code base. Probably not necessary. I can remove the whole `elemsMap` if you think that will make the code cleaner.

Reply:
Simplified in [8df15cb](https://github.com/opendatahub-io/opendatahub-operator/pull/668/commits/8df15cb8772536f149e6e24555c67d1b98e6f364)